### PR TITLE
add monitor linking and reuse faraday

### PR DESCRIPTION
**Pull request by grosser**
in preparation for adding more monitor logic ... I want 1 less abstraction layer and have logging for all http requests

@zendesk/compute @zendesk/samson 

Screen Shot 2019-06-19 at 7 10 05 PM

### CODE REVIEW

File: plugins/datadog/app/models/datadog_notification.rb [hunk 1]
```diff
@@ -20,34 +20,23 @@ def deliver(additional_tags: [], now: false)
         "error"
       end
 
-    event = Dogapi::Event.new(
-      body,
-      msg_title: @deploy.summary,
-      event_type: "deploy",
-      event_object: Digest::MD5.hexdigest("#{Time.new}|#{rand}"),
```
Comment (grosser) [inline plugins/datadog/app/models/datadog_notification.rb]: no idea why we did that, so removing when in doubt 🤷‍♂

File: plugins/datadog/app/models/datadog_notification.rb [hunk 2]
```diff
@@ -20,34 +19,19 @@ def deliver(additional_tags: [], now: false)
...
 
-  private
-
-  def body
-    "#{@deploy.user.email} deployed #{@deploy.short_reference} to #{@stage.name}"
-  end
-
-  def api_key
-    ENV["DATADOG_API_KEY"]
+    if response.status != 202
```
Comment (uthark) [inline plugins/datadog/app/models/datadog_notification.rb]: What about statuses
- 200 OK
- 201 Created?
https://docs.datadoghq.com/api/?lang=ruby#success-and-errors

Comment (grosser) [reply plugins/datadog/app/models/datadog_notification.rb]: yeah let's just say between 200 and 202 ... 202 is what datadog returns, but no need to hardcode this weirdness

**Comment by grosser** [comment]
found a 3 places where we were not logging request urls, so that's nice too :)

**Comment by yizhang-zen** [comment]
@grosser Are you going to add more monitor logic? Or is it ready for me to add monitors in Samson after this change rollout?

**Comment by grosser** [comment]
more coming

**Pull request merged**